### PR TITLE
Add client parameter in download record

### PR DESCRIPTION
**Pull request by Harsimran1**
https://gesundheitscloud.atlassian.net/browse/HC-1527

### CODE REVIEW

File: test/lib/taggingUtilsTest.js [hunk 1]
```diff
@@ -47,22 +47,20 @@ describe('taggingUtils', () => {
 
     taggingUtils.clientId = testVariables.clientId;
 
-    it('generateTags succeeds', (done) => {
+    it('verifies that generateTags succeeds', () => {
         const tags = taggingUtils.generateTags(fhirObject);
         expect(tags.length).to.equal(2);
         expect(tags[1]).to.equal('resourcetype=patient');
         expect(tags[0]).to.equal(taggingUtils.buildTag('client', testVariables.clientId));
-        done();
```
Comment (ibihim) [inline test/lib/taggingUtilsTest.js]: :+1:

File: test/lib/taggingUtilsTest.js [hunk 2]
```diff
@@ -47,22 +47,20 @@ describe('taggingUtils', () => {
...
 
-    it('buildTag returns correctly encoded tag', (done) => {
+    it('verifies that buildTag returns correctly encoded tag', () => {
         const tag = taggingUtils.buildTag(
             ';,/?:@&=+$#-_.!~*\'()ABC abc 123',
             ';,/?:@&=+$#-_.!~*\'()ABC abc 123');
         expect(tag).to.equal(
             '%3B%2C%2F%3F%3A%40%26%3D%2B%24%23%2d%5f%2e%21%7e%2a%27%28%29abc%20abc%20123='
             + '%3B%2C%2F%3F%3A%40%26%3D%2B%24%23%2d%5f%2e%21%7e%2a%27%28%29abc%20abc%20123');
-        done();
```
Comment (ibihim) [inline test/lib/taggingUtilsTest.js]: :+1:

File: test/lib/taggingUtilsTest.js [hunk 3]
```diff
@@ -72,7 +70,7 @@ describe('taggingUtils', () => {
         expect(tagValue).to.equal('1');
     });
 
-    it('returns correct annotaions when tagsList is passed', () => {
```
Comment (ibihim) [inline test/lib/taggingUtilsTest.js]: :+1:

File: test/services/fhirServiceTest.js [hunk 4]
```diff
@@ -216,7 +216,9 @@ describe('services/fhirService', () => {
         it('should resolve to Record, when called wit userId and recordId', (done) => {
             fhirService.downloadFhirRecord(testVariables.userId, testVariables.recordId)
                 .then((res) => {
-                    expect(res.body).to.deep.equal(fhirResources.documentReference);
+                    expect(res.body).to.deep.equal(
+                        JSON.parse(JSON.stringify(fhirResources.documentReference)),
```
Comment (ibihim) [inline test/services/fhirServiceTest.js]: Expect does not mutate the data, so why are you suggesting a deep clone here?

Comment (Harsimran1) [reply test/services/fhirServiceTest.js]: for some reason, it won't deep equal without doing this. But I figured out the issue that it was getting mutated in other tests in documentServiceTest. cloned the object there.

**Pull request merged**